### PR TITLE
feat(sdk): extract entity store for re-usability

### PR DESCRIPTION
**Pull request by chrishelgert**
- Extract the EntityStore for loading entities from the editor into separate modules
- Use it for resolving references
- Adjust the first steps for later message optimizations (from -> source, action -> method) for unified messaging

### CODE REVIEW

File: packages/visual-sdk/src/store/EditorEntityStore.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      const entry = this.entitiesMap.get(id);
+      if (!entry) {
+        missing.push(id);
+      }
+    }
+
+    return missing;
+  }
+
+  private async fetchEntity(ids: string[]): Promise<Array<Entry>>;
```
Comment (YvesRijckaert) [inline packages/visual-sdk/src/store/EditorEntityStore.ts:83]: I find this code a bit confusing, could explain what's happening here? Why do we have 3 times `fetchEntity`? Can we not have one generic function and then fetchAsset?

File: packages/visual-sdk/src/store/EntityStore.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    if (missingEntityIds.length) {
+      throw new Error(`Missing entity in the store (${missingEntityIds.join(',')})`);
+    }
+
+    return entity as Array<Asset | Entry>;
+  }
+
+  public async fetchAsset(id: string): Promise<Asset | undefined> {
```
Comment (YvesRijckaert) [inline packages/visual-sdk/src/store/EntityStore.ts:61]: should these methods be async? Aren't we just getting something from a map?

**Review by YvesRijckaert — APPROVED** [review]
Just two questions, looks good!

File: packages/visual-sdk/src/store/EditorEntityStore.ts (hunk 1 above)
Comment (chrishelgert) [reply packages/visual-sdk/src/store/EditorEntityStore.ts:83]: Ah that's for typing ([function overload](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads))- so you get the correct type

if you call it without  `isAsset` you get the `Entry` type
if you call it with `isAsset` you get the `Asset` type
and the third one is then the implementation, which needs to implement both overloads

**Pull request merged**